File: scripts/update-manifest.mjs
// Update manifest.json to point to the latest (alphabetically or by mtime) HTML file in /exports
// Usage: node scripts/update-manifest.mjs
import { promises as fs } from 'node:fs';
import path from 'node:path';
const root = process.cwd();
const exportsDir = path.join(root, 'exports');
const manifestPath = path.join(root, 'manifest.json');

async function main(){
  const entries = await fs.readdir(exportsDir, { withFileTypes: true });
  const htmls = await Promise.all(entries
    .filter(e => e.isFile() && e.name.toLowerCase().endsWith('.html'))
    .map(async e => {
      const full = path.join(exportsDir, e.name);
      const stat = await fs.stat(full);
      return { name: e.name, mtime: stat.mtimeMs };
    }));
  if(!htmls.length) throw new Error('No .html files in /exports');
  htmls.sort((a,b)=> b.mtime - a.mtime);
  const latest = htmls[0].name;

  let manifest = { title: 'Playbook', version: '1.0.0' };
  try{
    manifest = JSON.parse(await fs.readFile(manifestPath, 'utf8'));
  }catch{ /* ignore */ }

  manifest.export_html = `exports/${latest}`;
  await fs.writeFile(manifestPath, JSON.stringify(manifest, null, 2), 'utf8');
  console.log('manifest.json updated ->', manifest.export_html);
}

main().catch(err=>{ console.error(err); process.exit(1); });

File: scripts/update_manifest.py
#!/usr/bin/env python3
"""
Update manifest.json to point to the most recent .html file in ./exports/
Selection is by modification time (newest wins).

Usage:
  python scripts/update_manifest.py
"""
from __future__ import annotations
import json, os, sys, time
from pathlib import Path

ROOT = Path(__file__).resolve().parents[1]
EXPORTS = ROOT / "exports"
MANIFEST = ROOT / "manifest.json"

def latest_export() -> Path:
    htmls = [p for p in EXPORTS.glob("*.html") if p.is_file()]
    if not htmls:
        raise SystemExit("No .html files found in ./exports")
    htmls.sort(key=lambda p: p.stat().st_mtime, reverse=True)
    return htmls[0]

def main():
    EXPORTS.mkdir(parents=True, exist_ok=True)
    latest = latest_export()
    manifest = {}
    if MANIFEST.exists():
        try:
            manifest = json.loads(MANIFEST.read_text("utf-8"))
        except Exception:
            manifest = {}
    manifest.setdefault("title", "Playbook")
    manifest["export_html"] = f"exports/{latest.name}"
    MANIFEST.write_text(json.dumps(manifest, indent=2), "utf-8")
    print("manifest.json ->", manifest["export_html"])

if __name__ == "__main__":
    main()
